File: question1.rb
#Read the number n and Design a program that prints the numbers from 1 to n.
#But for multiples of three print "Boats" instead of the number.
#For the multiples of five print "Group". 
#For numbers which are multiples of both three and five print "BoatsGroup"
class Question1

  attr_reader :arr

  def initialize (n)
    @arr = Array.new(n) { |i| process(i) }
  end

private

  def process (i)
    i+=1
    if i%3 == 0
      if i%5 == 0
        "BoatsGroup"
      else
        "Boats"
      end
    else
      if i%5 == 0
        "Group"
      else
        "#{i}"
      end    
    end
  end

end

File: Question1.java
import java.io.*;
import java.util.*;


public class Question1 {

  private String[] arr;

  public String[] getArr(){
    return arr;
  }

  Question1(int n){
    arr = new String[n];
    for(int i=1; i <= n; i++){
      if (i%3 == 0)
        if (i%5 == 0)
          arr[i-1]="BoatsGroup";
        else
          arr[i-1]="Boats";
      else
        if (i%5 == 0)
          arr[i-1]="Group";
        else
          arr[i-1]=String.format("%s",i);
    }
  }

  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    int n = in.nextInt();
    Question1 q = new Question1(n);
    String[] arr = q.getArr();
    for(int i=0; i<n; i++){
      System.out.println(arr[i]);
    }
  }
}
